rest-service/manager_rest/test/endpoints/test_deployment_labels_dependencies.py

```diff
@@ -1,613 +1,420 @@
-           '.verify_attaching_deployment_to_parents')
-    def test_deployment_with_empty_labels(self,
-                                          verify_parents_mock,
-                                          handle_labels_graph_mock):
-        self.put_deployment('deployment_with_no_labels')
-        verify_parents_mock.assert_not_called()
-        handle_labels_graph_mock.assert_not_called()
-
-    @patch('manager_rest.resource_manager.ResourceManager'
-           '.handle_deployment_labels_graph')
-    @patch('manager_rest.resource_manager.ResourceManager'
-           '.verify_attaching_deployment_to_parents')
-    def test_deployment_with_non_parent_labels(self,
-                                               verify_parents_mock,
-                                               handle_labels_graph_mock):
-        self.put_deployment_with_labels([{'env': 'aws'}, {'arch': 'k8s'}])
-        verify_parents_mock.assert_not_called()
-        handle_labels_graph_mock.assert_not_called()
+    def _deployment(self, **kwargs):
+        dep_attrs = {
+            'blueprint': self.blueprint,
+            'tenant': self.tenant,
+            'creator': self.user
+        }
+        dep_attrs.update(kwargs)
+        dep = models.Deployment(**dep_attrs)
+        db.session.add(dep)
+        return dep
 
     def test_deployment_with_single_parent_label(self):
-        self.put_deployment('parent')
-        self.put_deployment_with_labels([{'csys-obj-parent': 'parent'}])
-
-        # deployment response
-        deployment = self.client.deployments.get('parent')
-        self.assertEqual(deployment.sub_services_count, 1)
-        self.assertEqual(deployment.sub_environments_count, 0)
-
-    def test_deploy_blueprint_with_invalid_parent_id_on_dsl(self):
-        with self.assertRaisesRegex(
-                CloudifyClientError,
-                'using label `csys-obj-parent` that does not exist'):
-            self.put_deployment(
-                blueprint_id='bp1',
-                blueprint_file_name='blueprint_with_invalid_parent_labels.yaml'
+        parent = self._deployment(id='parent')
+        self._deployment(id='dep')
+
+        self.client.deployments.set_attributes(
+            'dep',
+            labels=[{'csys-obj-parent': 'parent'}]
+        )
+        assert parent.sub_services_count == 1
+        assert parent.sub_environments_count == 0
+
+    def test_deploy_blueprint_with_invalid_parent_id(self):
+        self._deployment(id='dep')
+        with self.assertRaisesRegex(CloudifyClientError, 'does not exist'):
+            self.client.deployments.set_attributes(
+                'dep',
+                labels=[{'csys-obj-parent': 'parent'}]
             )
 
-    def test_upload_blueprint_with_valid_parent_id_on_dsl(self):
-        self.put_deployment('valid-id')
-        self.put_blueprint(
-            blueprint_id='bp1',
-            blueprint_file_name='blueprint_with_valid_parent_labels.yaml'
-        )
```

Comment:
the `on_dsl` tests, which test stuff based on a blueprint with an actual .yaml file, NEED TO GTFO.
If you want to test the whole flow, from a .yaml, to behaviour, there's inte-tests for that.

On this level, we just check the behaviour of specific rest endpoints.

What happens with a yaml file, well, that's to be defined by the dsl-parser, and the create-dep-env workflow.
In another commit in this PR, I add tests for create-dep-env, who check that the labels from the dsl, are passed to the restservice. And since we know they're indeed passed in, here we just need to check the behaviour of that endpoint.